docs/cgm.rst

```diff
@@ -0,0 +1,120 @@
+.. highlight:: rest
+
+***********
+CGM Classes
+***********
+
+.. index:: CGM
+
+Notebooks
+=========
+
+.. toctree::
+   :maxdepth: 1
+
+   CGM Examples <CGM_examples>
+
+Overview
+========
+
+There are a series of classes designed for CGM analyses.
+We describe each in turn.
+
+CGM
+===
+
+This class is intended to provide a physical representation
+of the CGM.  A Galaxy object is required for instantiation.
+The key attributes are tabulated below.  This includes several
+which may be controversial, e.g. the physical extent of the CGM,
+and a description of the baryons in `phases'.
+This is the least developed of the classes.
+
+=========   ============== ============================================
+Attribute   Type           Description
+=========   ============== ============================================
+galaxy      Galaxy object  Describes the galaxy hosting the CGM
+rlim        Quantity       Physical extent of the CGM (e.g. 300 kpc)
+phases      dict           Intended to organize the properties of various CGM phases, e.g. mass, metallicity
+cgm_abs     list           List of CGMAbsSys classes
+=========   ============== ============================================
+
+Instantiation
+-------------
+
+The only requirement is a Galaxy object.::
+
+   radec = (125*u.deg, 45.2*u.deg)
+   gal = Galaxy(radec,z=0.3)
+   cgm = CGM(gal)
+
+CGMAbsSys
+=========
+
+This class enables an absorption-line analysis of a CGM.  This has been
+the primary approach to CGM analysis to date.  This class requires
+both a Galaxy object and :doc:`igmsys` object for instantiation.
+
+=========   ================= ============================================
+Attribute   Type              Description
+=========   ================= ============================================
+galaxy      Galaxy object     Describes the galaxy hosting the CGM
+igm_sys     IGMSystem         IGM system object to describe absorption on the line-of-sight
+cosmo       astropy.cosmology Cosmological model; defaults to WMAP9
+rho         Quantity          Impact parameter from galaxy to sightline
+=========   ================= ============================================
+
+Instantiation
+-------------
+
+One must invoke CGMAbsSys with both a Galaxy and IGMSystem object.::
+
+   radec_qso = (125*u.deg, 45.203*u.deg)
+   igmsys = IGMSystem('CGM', radec_qso, gal.z, [-500,500]*u.km/u.s)
+   #
+   radec = (125*u.deg, 45.2*u.deg)
+   gal = Galaxy(radec,z=0.3)
+   #
+   cgmabs = CGMAbsSys(gal,igmsys)
+
+CGMAbsSurvey
+============
+
+This class organizes a survey of CGMAbsSys objects, e.g. COS-Halos.
+
+=========   ================= ============================================
+Attribute   Type              Description
+=========   ================= ============================================
+survey      str               Name of the survey, e.g. COS-Halos
+ref         str               References for the survey
+cgm_abs     list              list of CGMAbsSys objects
```

Comment:
No, the intent would be to have a CGMSurvey object.  
But that doesn't exist yet nor do I think that I have an
application for it yet.